modin/core/dataframe/base/partitioning/axis_partition.py

```diff
@@ -88,10 +94,6 @@ def shuffle(self, func, lengths, **kwargs):
         """
         pass
 
-    # Child classes must have these in order to correctly subclass.
-    instance_type = None
-    partition_type = None
-
     def _wrap_partitions(self, partitions):
```

Comment:
All the methods that have implementations should be moved to PandasDataframeAxisPartition.

Reply:
Why? The only implemented method that seems to depend on pandas is `force_materialization`, the rest looks pretty generic and could be preserved in the base class, Am I missing something?

Reply:
An abstract class shouldn't have implementations, just declare an interface.

Reply:
Are you sure we should treat an <i>abstract class</i> as a pure <i>interface</i> here? If so, then we have to rename `BaseDataframePartition -> IDataFramePartition/DataFramePartitionInterface` as it's not a base class anymore. 

Personally, I don't like this approach as it will force us to declare yet another class when we would actually need to create a base partition class (which would bloat our namespace).

<details><summary>Example</summary>

```python
# ====== Pure interface approach =======
class IClass(ABC):
    @abstractmethod
    def implementation_specific_method(self):
        """BaseClass API that has no common implementation."""
        pass

class BaseClass(IClass): # This class is still abstract
    def generic_method(self):
        """
        Still BaseClass API but already has an implementation that is suitable for
        every descendant. Could be overridden in order to achieve better performance.
        """
        res = self.implementation_specific_method()
        return common_postprocessing(res)

# ====== No interface approach =======
class BaseClass(ABC):
    @abstractmethod
    def implementation_specific_method(self):
        """BaseClass API that has no common implementation."""
        pass

    def generic_method(self):
        """
        Still BaseClass API but already has an implementation that is suitable for
        every descendant. Could be overridden in order to achieve better performance.
        """
        res = self.implementation_specific_method()
        return common_postprocessing(res)
```

</details>

Reply:
I also personally like abstract classes in С++ style.